[libpas] Centralize most logging flags

Achierius opened the pull request:
#### dde95ad41f4e3b5bf46336bf9e47433818a1c569
<pre>
[libpas] Centralize most logging flags
<a href="https://bugs.webkit.org/show_bug.cgi?id=280975">https://bugs.webkit.org/show_bug.cgi?id=280975</a>
<a href="https://rdar.apple.com/137427586">rdar://137427586</a>

Reviewed by Mark Lam.

Previously, these flags had to be enabled one-by-one for each file (and
often each function) you wanted to log from. Now the most common -- at
least in my experience -- collections of log-prints can be enabled by
changing PAS_LOG_LEVEL to include whichever levels should be logged.

I didn&apos;t cover all of the verbose prints in libpas, as some of the
others are particularly spammy or would otherwise need some cleanup
before they could be used. I also left the existing verbose flags
in-place in case they&apos;re part of anyone&apos;s workflow.

* Source/bmalloc/libpas/src/libpas/jit_heap.c:
(jit_heap_add_fresh_memory):
(jit_heap_try_allocate):
(jit_heap_shrink):
(jit_heap_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator.c:
(pas_bitfit_allocator_commit_view):
(pas_bitfit_allocator_finish_failing):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator_inlines.h:
(pas_bitfit_allocator_try_allocate):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_directory.c:
(pas_bitfit_directory_construct):
(pas_bitfit_directory_get_first_free_view):
(pas_bitfit_directory_take_last_empty):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_heap.c:
(pas_bitfit_heap_select_variant):
(pas_bitfit_heap_construct_and_insert_size_class):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_page.c:
(pas_bitfit_page_construct):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_page_inlines.h:
(pas_bitfit_page_allocation_satisfies_alignment):
(pas_bitfit_page_finish_allocation):
(pas_bitfit_page_allocate):
(pas_bitfit_page_deallocate_with_page_impl):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_view.c:
(pas_bitfit_view_create):
(compute_summary):
* Source/bmalloc/libpas/src/libpas/pas_bootstrap_free_heap.c:
(bootstrap_source_allocate_aligned):
* Source/bmalloc/libpas/src/libpas/pas_bootstrap_heap_page_provider.c:
(pas_bootstrap_heap_page_provider):
* Source/bmalloc/libpas/src/libpas/pas_compact_bootstrap_free_heap.c:
(compact_bootstrap_source_allocate_aligned):
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_deallocate.c:
(pas_try_deallocate_slow_no_cache):
* Source/bmalloc/libpas/src/libpas/pas_deallocate.h:
(pas_try_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_enumerate_bitfit_heaps.c:
(view_callback):
* Source/bmalloc/libpas/src/libpas/pas_enumerate_segregated_heaps.c:
(record_page_objects):
(enumerate_exclusive_view):
(enumerate_shared_view):
(enumerate_partial_view):
(consider_allocator):
* Source/bmalloc/libpas/src/libpas/pas_fast_large_free_heap.c:
(pas_fast_large_free_heap_try_allocate):
(pas_fast_large_free_heap_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_heap.c:
(pas_heap_create):
* Source/bmalloc/libpas/src/libpas/pas_heap_inlines.h:
(pas_heap_ensure_size_directory_for_size):
* Source/bmalloc/libpas/src/libpas/pas_heap_table.h:
(pas_heap_table_has_index):
* Source/bmalloc/libpas/src/libpas/pas_immortal_heap.c:
(pas_immortal_heap_allocate_with_manual_alignment):
(pas_immortal_heap_allocate_with_alignment):
* Source/bmalloc/libpas/src/libpas/pas_large_free_heap_helpers.c:
(large_utility_aligned_allocator):
(pas_large_free_heap_helpers_try_allocate_with_alignment):
(pas_large_free_heap_helpers_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_large_heap.c:
(allocate_impl):
* Source/bmalloc/libpas/src/libpas/pas_large_map.c:
(pas_large_map_add):
(pas_large_map_take):
* Source/bmalloc/libpas/src/libpas/pas_local_allocator.c:
(pas_local_allocator_construct):
(stop_impl):
(pas_local_allocator_stop):
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
(pas_local_allocator_scan_bits_to_set_up_free_bits):
(pas_local_allocator_prepare_to_allocate):
(pas_local_allocator_set_up_primordial_bump):
(pas_local_allocator_start_allocating_in_primordial_partial_view):
(pas_local_allocator_bless_primordial_partial_view_before_stopping):
(pas_local_allocator_refill_with_known_config):
(pas_local_allocator_return_memory_to_page_for_role):
(pas_local_allocator_try_allocate_with_free_bits):
(pas_local_allocator_try_allocate_inline_cases):
(pas_local_allocator_try_allocate_out_of_line_cases):
(pas_local_allocator_try_allocate_slow_impl):
(pas_local_allocator_try_allocate_inline_only):
(pas_local_allocator_try_allocate):
* Source/bmalloc/libpas/src/libpas/pas_page_malloc.c:
(pas_page_malloc_try_allocate_without_deallocating_padding):
(decommit_impl):
* Source/bmalloc/libpas/src/libpas/pas_segregated_directory.c:
(pas_segregated_directory_get_use_epoch):
(pas_segregated_directory_get_sharing_payload):
(pas_segregated_directory_view_did_become_empty_at_index):
* Source/bmalloc/libpas/src/libpas/pas_segregated_directory_inlines.h:
(pas_segregated_directory_iterate_iterate_callback):
(pas_segregated_directory_iterate_forward):
(pas_segregated_directory_iterate_forward_to_take_first_eligible):
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view.c:
(pas_segregated_exclusive_view_create):
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view_inlines.h:
(pas_segregated_exclusive_view_did_start_allocating):
(pas_segregated_exclusive_view_did_stop_allocating):
(pas_segregated_exclusive_view_note_eligibility):
* Source/bmalloc/libpas/src/libpas/pas_segregated_heap.c:
(max_object_size_for_page_config):
(max_segregated_object_size_for_heap):
(max_bitfit_object_size_for_heap):
(medium_directory_tuple_for_index_impl):
(pas_segregated_heap_medium_directory_tuple_for_index):
(pas_segregated_heap_medium_allocator_index_for_index):
(ensure_size_lookup):
(pas_segregated_heap_ensure_allocator_index):
(compute_ideal_object_size):
(check_part_of_all_heaps_callback):
(pas_segregated_heap_ensure_size_directory_for_size):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.c:
(pas_segregated_page_switch_lock_slow):
(pas_segregated_page_construct):
(pas_segregated_page_note_emptiness):
(pas_segregated_page_take_empty_granules):
(pas_segregated_page_commit_fully):
(verify_granules_live_object_callback):
(pas_segregated_page_verify_granules):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.h:
(pas_segregated_page_number_of_objects):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
(pas_segregated_page_switch_lock_impl):
(pas_segregated_page_deallocate_with_page):
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.c:
(pas_segregated_partial_view_create):
(pas_segregated_partial_view_note_eligibility):
(pas_segregated_partial_view_set_is_in_use_for_allocation):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.c:
(pas_segregated_shared_handle_note_emptiness):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h:
(pas_segregated_shared_handle_partial_view_ptr_for_index):
(pas_segregated_shared_handle_partial_view_for_object):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.c:
(find_first_eligible_consider_view):
(pas_segregated_shared_page_directory_find_first_eligible):
(take_last_empty_consider_view):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.c:
(compute_summary_for_each_live_object_callback):
(compute_summary):
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory.c:
(pas_segregated_size_directory_create):
(take_first_eligible_direct_create_new_view_callback):
(pas_segregated_size_directory_take_first_eligible):
(take_last_empty_consider_view):
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory_inlines.h:
(pas_segregated_size_directory_num_allocator_indices_for_config):
(pas_segregated_size_directory_take_first_eligible_impl):
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.c:
(for_each_live_object):
(should_be_eligible):
(pas_segregated_view_should_be_eligible):
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
(pas_segregated_view_will_start_allocating):
(pas_segregated_view_did_stop_allocating):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate.h:
(pas_try_allocate_impl_inline_only):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_common.h:
(pas_try_allocate_common_impl_fast):
(pas_try_allocate_common_impl_slow):
(pas_try_allocate_common_impl):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_intrinsic.h:
(pas_try_allocate_intrinsic_impl_casual_case):
(pas_try_allocate_intrinsic_impl_inline_only):
* Source/bmalloc/libpas/src/libpas/pas_try_reallocate.h:
(pas_try_allocate_for_reallocate_and_copy):
(pas_try_reallocate_primitive_allocate_callback):

Canonical link: <a href="https://commits.webkit.org/285358@main">https://commits.webkit.org/285358@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0797dc3b1dc9124dedfc89393dd94a78f64a72b9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72389 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51810 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25177 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23598 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59614 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23420 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/76568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/15533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75456 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/46916 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62340 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/43568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19809 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21948 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/65518 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/65452 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20167 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78237 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/71643 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16630 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/19306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16678 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62355 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/64741 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13003 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/6637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/93424 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11108 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47608 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/20558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/48677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/49965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48420 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/bmalloc/libpas/src/libpas/jit_heap.c:
```diff
@@ -54,12 +54,15 @@ void jit_heap_add_fresh_memory(pas_range range)
 
-    if (verbose)
-        pas_log("JIT heap at %p...%p\n", (void*)range.begin, (void*)range.end);
+    if (verbose || PAS_SHOULD_LOG(PAS_LOG_JIT_HEAPS))
```
Why not just change the definition of `verbose` to be:
```
static const bool verbose = false || PAS_SHOULD_LOG(PAS_LOG_JIT_HEAPS);
```

Then, you don't have to make is many edits, and the code is also more compact and easier to read.

Achierius replied:
Good point, will change

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator.c:
```diff
@@ -175,8 +175,11 @@ bool pas_bitfit_allocator_commit_view(pas_bitfit_view* view,
-        if (verbose)
+        if (verbose) {
+            // Without this comment the style checker gets caught in a loop here
+            // for some reason, objecting both to including and excluding braces
```
Is this still needed given that the pre-existing code didn't need it?

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_bitfit_heap.c:
```diff
@@ -83,9 +83,10 @@ pas_bitfit_variant_selection pas_bitfit_heap_select_variant(size_t requested_obj
             continue;
 
-        if (verbose)
+        if (verbose) {
```
Ditto with the { }

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_bitfit_heap.c:
```diff
@@ -130,10 +131,11 @@ void pas_bitfit_heap_construct_and_insert_size_class(pas_bitfit_heap* heap,
     
     pas_bitfit_variant_selection best;
     pas_compact_atomic_bitfit_size_class_ptr* insertion_point;
+
```
nit: Please remove.  I don't see how this empty line adds value.

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_large_free_heap_helpers.c:
```diff
@@ -164,7 +164,7 @@ void pas_large_free_heap_helpers_deallocate(
     if (pas_large_utility_free_heap_talks_to_large_sharing_pool) {
         if (verbose) {
-            pas_log("freeing %p...%p.\n",
+            pas_log("pas large free heap freeing %p...%p.\n",
```
Why add the "pas " prefix?  I don't see this being done elsewhere.  Let's be consistent.  I suggest leaving it out unless there's some strong reason why we need it.

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_large_heap.c:
```diff
@@ -117,9 +117,8 @@ static pas_allocation_result allocate_impl(pas_large_heap* heap,
-        printf("Cartesian tree minimum = %p\n", pas_cartesian_tree_minimum(&heap->free_heap.tree));
-        printf("Num mapped bytes = %zu\n", heap->free_heap.num_mapped_bytes);
+        printf("pas large heap allocating large object of size %zu\n", *size);
+        printf("pas large heap cartesian tree minimum = %p, num mapped bytes = %zu\n", pas_cartesian_tree_minimum(&heap->free_heap.tree), heap->free_heap.num_mapped_bytes);
```
Ditto with "pas " prefix.

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_large_heap.c:
```diff
@@ -134,7 +133,7 @@ static pas_allocation_result allocate_impl(pas_large_heap* heap,
 
     if (verbose)
-        pas_log("Committing the memory we allocated starting at %p.\n", (void*)result.begin);
+        pas_log("pas large heap committing the memory we allocated starting at %p.\n", (void*)result.begin);
```
Ditto with "pas " prefix.

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_config.h:
```diff
@@ -28,6 +28,21 @@
+#define PAS_LOG_BITFIT_HEAPS (1 << 4)
+#define PAS_LOG_LARGE_HEAPS (1 << 5)
+#define PAS_LOG_JIT_HEAPS (1 << 6)
+#define PAS_LOG_SURFACE_ALLOCATIONS (1 << 7)  /* Heap-type agnostic */
```
nit: It feels like the term SURFACE sort of implies something.  How about naming this as something like PAS_LOG_OTHER_ALLOCATIONS or PAS_LOG_HEAP_TYPE_AGNOSTIC_ALLOCATIONS or just PAS_LOG_OTHER  instead?  I prefer PAS_LOG_OTHER as it's short, most generic, and applies to all logging, not just allocations ... and you do seem to be using it that way.

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
```diff
@@ -397,8 +397,10 @@ pas_segregated_page_deallocate_with_page(pas_segregated_page* page,
 
-    if (verbose)
+    if (verbose
+        || PAS_SHOULD_LOG(PAS_LOG_SEGREGATED_HEAPS)) {
```
This is no longer needed.

MenloDorian reviewed (approved):
r=m with suggestions and a few things to follow up on.

Achierius committed:
[libpas] Centralize most logging flags
https://bugs.webkit.org/show_bug.cgi?id=280975
rdar://137427586

Reviewed by Mark Lam.

Previously, these flags had to be enabled one-by-one for each file (and
often each function) you wanted to log from. Now the most common -- at
least in my experience -- collections of log-prints can be enabled by
changing PAS_LOG_LEVEL to include whichever levels should be logged.

I didn't cover all of the verbose prints in libpas, as some of the
others are particularly spammy or would otherwise need some cleanup
before they could be used. I also left the existing verbose flags
in-place in case they're part of anyone's workflow.

* Source/bmalloc/libpas/src/libpas/jit_heap.c:
(jit_heap_add_fresh_memory):
(jit_heap_try_allocate):
(jit_heap_shrink):
(jit_heap_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator.c:
(pas_bitfit_allocator_commit_view):
(pas_bitfit_allocator_finish_failing):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator_inlines.h:
(pas_bitfit_allocator_try_allocate):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_directory.c:
(pas_bitfit_directory_construct):
(pas_bitfit_directory_get_first_free_view):
(pas_bitfit_directory_take_last_empty):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_heap.c:
(pas_bitfit_heap_select_variant):
(pas_bitfit_heap_construct_and_insert_size_class):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_page.c:
(pas_bitfit_page_construct):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_page_inlines.h:
(pas_bitfit_page_allocation_satisfies_alignment):
(pas_bitfit_page_finish_allocation):
(pas_bitfit_page_allocate):
(pas_bitfit_page_deallocate_with_page_impl):
* Source/bmalloc/libpas/src/libpas/pas_bitfit_view.c:
(pas_bitfit_view_create):
(compute_summary):
* Source/bmalloc/libpas/src/libpas/pas_bootstrap_free_heap.c:
(bootstrap_source_allocate_aligned):
* Source/bmalloc/libpas/src/libpas/pas_bootstrap_heap_page_provider.c:
(pas_bootstrap_heap_page_provider):
* Source/bmalloc/libpas/src/libpas/pas_compact_bootstrap_free_heap.c:
(compact_bootstrap_source_allocate_aligned):
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_deallocate.c:
(pas_try_deallocate_slow_no_cache):
* Source/bmalloc/libpas/src/libpas/pas_deallocate.h:
(pas_try_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_enumerate_bitfit_heaps.c:
(view_callback):
* Source/bmalloc/libpas/src/libpas/pas_enumerate_segregated_heaps.c:
(record_page_objects):
(enumerate_exclusive_view):
(enumerate_shared_view):
(enumerate_partial_view):
(consider_allocator):
* Source/bmalloc/libpas/src/libpas/pas_fast_large_free_heap.c:
(pas_fast_large_free_heap_try_allocate):
(pas_fast_large_free_heap_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_heap.c:
(pas_heap_create):
* Source/bmalloc/libpas/src/libpas/pas_heap_inlines.h:
(pas_heap_ensure_size_directory_for_size):
* Source/bmalloc/libpas/src/libpas/pas_heap_table.h:
(pas_heap_table_has_index):
* Source/bmalloc/libpas/src/libpas/pas_immortal_heap.c:
(pas_immortal_heap_allocate_with_manual_alignment):
(pas_immortal_heap_allocate_with_alignment):
* Source/bmalloc/libpas/src/libpas/pas_large_free_heap_helpers.c:
(large_utility_aligned_allocator):
(pas_large_free_heap_helpers_try_allocate_with_alignment):
(pas_large_free_heap_helpers_deallocate):
* Source/bmalloc/libpas/src/libpas/pas_large_heap.c:
(allocate_impl):
* Source/bmalloc/libpas/src/libpas/pas_large_map.c:
(pas_large_map_add):
(pas_large_map_take):
* Source/bmalloc/libpas/src/libpas/pas_local_allocator.c:
(pas_local_allocator_construct):
(stop_impl):
(pas_local_allocator_stop):
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
(pas_local_allocator_scan_bits_to_set_up_free_bits):
(pas_local_allocator_prepare_to_allocate):
(pas_local_allocator_set_up_primordial_bump):
(pas_local_allocator_start_allocating_in_primordial_partial_view):
(pas_local_allocator_bless_primordial_partial_view_before_stopping):
(pas_local_allocator_refill_with_known_config):
(pas_local_allocator_return_memory_to_page_for_role):
(pas_local_allocator_try_allocate_with_free_bits):
(pas_local_allocator_try_allocate_inline_cases):
(pas_local_allocator_try_allocate_out_of_line_cases):
(pas_local_allocator_try_allocate_slow_impl):
(pas_local_allocator_try_allocate_inline_only):
(pas_local_allocator_try_allocate):
* Source/bmalloc/libpas/src/libpas/pas_page_malloc.c:
(pas_page_malloc_try_allocate_without_deallocating_padding):
(decommit_impl):
* Source/bmalloc/libpas/src/libpas/pas_segregated_directory.c:
(pas_segregated_directory_get_use_epoch):
(pas_segregated_directory_get_sharing_payload):
(pas_segregated_directory_view_did_become_empty_at_index):
* Source/bmalloc/libpas/src/libpas/pas_segregated_directory_inlines.h:
(pas_segregated_directory_iterate_iterate_callback):
(pas_segregated_directory_iterate_forward):
(pas_segregated_directory_iterate_forward_to_take_first_eligible):
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view.c:
(pas_segregated_exclusive_view_create):
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view_inlines.h:
(pas_segregated_exclusive_view_did_start_allocating):
(pas_segregated_exclusive_view_did_stop_allocating):
(pas_segregated_exclusive_view_note_eligibility):
* Source/bmalloc/libpas/src/libpas/pas_segregated_heap.c:
(max_object_size_for_page_config):
(max_segregated_object_size_for_heap):
(max_bitfit_object_size_for_heap):
(medium_directory_tuple_for_index_impl):
(pas_segregated_heap_medium_directory_tuple_for_index):
(pas_segregated_heap_medium_allocator_index_for_index):
(ensure_size_lookup):
(pas_segregated_heap_ensure_allocator_index):
(compute_ideal_object_size):
(check_part_of_all_heaps_callback):
(pas_segregated_heap_ensure_size_directory_for_size):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.c:
(pas_segregated_page_switch_lock_slow):
(pas_segregated_page_construct):
(pas_segregated_page_note_emptiness):
(pas_segregated_page_take_empty_granules):
(pas_segregated_page_commit_fully):
(verify_granules_live_object_callback):
(pas_segregated_page_verify_granules):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.h:
(pas_segregated_page_number_of_objects):
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
(pas_segregated_page_switch_lock_impl):
(pas_segregated_page_deallocate_with_page):
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.c:
(pas_segregated_partial_view_create):
(pas_segregated_partial_view_note_eligibility):
(pas_segregated_partial_view_set_is_in_use_for_allocation):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.c:
(pas_segregated_shared_handle_note_emptiness):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h:
(pas_segregated_shared_handle_partial_view_ptr_for_index):
(pas_segregated_shared_handle_partial_view_for_object):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.c:
(find_first_eligible_consider_view):
(pas_segregated_shared_page_directory_find_first_eligible):
(take_last_empty_consider_view):
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.c:
(compute_summary_for_each_live_object_callback):
(compute_summary):
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory.c:
(pas_segregated_size_directory_create):
(take_first_eligible_direct_create_new_view_callback):
(pas_segregated_size_directory_take_first_eligible):
(take_last_empty_consider_view):
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory_inlines.h:
(pas_segregated_size_directory_num_allocator_indices_for_config):
(pas_segregated_size_directory_take_first_eligible_impl):
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.c:
(for_each_live_object):
(should_be_eligible):
(pas_segregated_view_should_be_eligible):
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
(pas_segregated_view_will_start_allocating):
(pas_segregated_view_did_stop_allocating):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate.h:
(pas_try_allocate_impl_inline_only):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_common.h:
(pas_try_allocate_common_impl_fast):
(pas_try_allocate_common_impl_slow):
(pas_try_allocate_common_impl):
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_intrinsic.h:
(pas_try_allocate_intrinsic_impl_casual_case):
(pas_try_allocate_intrinsic_impl_inline_only):
* Source/bmalloc/libpas/src/libpas/pas_try_reallocate.h:
(pas_try_allocate_for_reallocate_and_copy):
(pas_try_reallocate_primitive_allocate_callback):

Canonical link: https://commits.webkit.org/285358@main
